### PR TITLE
Updating to elasticsearch 5.3.0

### PR DESCRIPTION
**Pull request by telser**
A few imports moved, and a few other things were renamed. Tests passed locally, but we'll see about the CI builds. Let me know if there is anything I need to change.

### CODE REVIEW

File: elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/index/IndexImplicits.scala [hunk 1]
```diff
@@ -23,7 +23,11 @@ trait IndexImplicits extends IndexShowImplicits {
       }
 
       val params = scala.collection.mutable.Map.empty[String, String]
-      request.opType.foreach(opType => params.put("op_type", opType.lowercase))
+      request.createOnly.foreach(createOnly => 
+        if(createOnly) {
+          params.put("op_type", "create")
```
Comment (telser) [inline elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/index/IndexImplicits.scala]: What is the name for the field in the Elasticsearch codebase now? "createOnly" But the param on the http client is still called "op_type". :/

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/indexes/IndexDefinition.scala [hunk 2]
```diff
@@ -3,13 +3,12 @@ package com.sksamuel.elastic4s.indexes
...
 import com.sksamuel.elastic4s.{FieldValue, FieldsMapper, IndexAndType, Indexable}
 import com.sksamuel.exts.OptionImplicits._
-import org.elasticsearch.action.index.IndexRequest.OpType
 import org.elasticsearch.action.support.WriteRequest.RefreshPolicy
 import org.elasticsearch.index.VersionType
 
 case class IndexDefinition(indexAndType: IndexAndType,
                            id: Option[Any] = None,
-                           opType: Option[OpType] = None,
+                           createOnly: Option[Boolean] = None,
```
Comment (telser) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/indexes/IndexDefinition.scala]: Just mirroring the name from Elasticsearch.

File: elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/search/SearchDslTest.scala [hunk 3]
```diff
@@ -188,7 +188,7 @@ class SearchDslTest extends FlatSpec with MockitoSugar with JsonSugar with OneIn
       hasParentQuery("sometype") query {
         "coldplay"
       } scoreMode true boost 1.2
-    } searchType SearchType.QUERY_AND_FETCH
```
Comment (telser) [inline elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/search/SearchDslTest.scala]: QUERY_AND_FETCH was removed, is there another consequence of this change here?

File: elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/search/SearchDslTest.scala [hunk 4]
```diff
@@ -431,7 +431,7 @@ class SearchDslTest extends FlatSpec with MockitoSugar with JsonSugar with OneIn
 
   it should "generate correct json for geo sort" in {
     val req = search("music") types "bands" sortBy {
-      geoSort("location").points("ABCDEFG").sortMode(SortMode.MAX).geoDistance(GeoDistance.SLOPPY_ARC)
```
Comment (telser) [inline elastic4s-tests/src/test/scala/com/sksamuel/elastic4s/search/SearchDslTest.scala]: SLOPPY_ARC and FACTOR were removed upstream.

**Comment by telser** [comment]
I left several comments on the changes for this, which I thought showed up on this page, but apparently not.

**Comment by sksamuel** [comment]
I think it looks great.

**Pull request merged**